### ISSUE TITLE
There may be a bug in the msvc detection part of rustc

### ISSUE THREAD

**Issue by 834772509** [rust-lang/rust#86835]
I created a new project using cargo, and then created a build.rs file with only an empty main function.
Then I execute `\VC\Auxiliary\Build\vcvarsall.bat x86` and then execute `cargo build --release --target=i686-pc-windows-msvc` and it will report an error: "The library computer type "x86" conflicts with the target computer type "x64" "
I think there may be a bug in the msvc detection part of Rustc.

**Comment by ChrisDenton** [comment]
Thanks for the report! Could you try with the nightly compiler in case this is fixed in the next version? If you're using rustup you can use nightly like this:
```
> rustup install nightly
> cargo +nightly build --release --target=i686-pc-windows-msvc
```

**Comment by 834772509** [comment]
ok, I tried the nightly version, it can be built normally, and I look forward to stabilizing this function as soon as possible

**Comment by JohnTitor** [comment]
This is fixed by #85649 and the next stable version (1.54) should have that fix. Thanks!
